controller/area_blackbox_test.go

```diff
@@ -60,190 +59,204 @@ func (rest *TestAreaREST) UnSecuredController() (*goa.Service, *AreaController)
-}
-
-func (rest *TestAreaREST) TestFailCreateChildAreaWithInvalidsParent() {
-	// given
-	createChildAreaPayload := newCreateChildAreaPayload("TestFailCreateChildAreaWithInvalidsParent")
-	svc, ctrl := rest.SecuredController()
-	// when/then
-	test.CreateChildAreaNotFound(rest.T(), svc.Context, svc, ctrl, uuid.NewV4().String(), createChildAreaPayload)
-}
-
-func (rest *TestAreaREST) TestFailCreateChildAreaNotAuthorized() {
-	// given
-	_, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	parentID := parentArea.ID
-	createChildAreaPayload := newCreateChildAreaPayload("TestFailCreateChildAreaNotAuthorized")
-	svc, ctrl := rest.UnSecuredController()
-	// when/then
-	test.CreateChildAreaUnauthorized(rest.T(), svc.Context, svc, ctrl, parentID.String(), createChildAreaPayload)
-}
-
-func (rest *TestAreaREST) TestFailValidationAreaNameLength() {
-	// given
-	ci := newCreateChildAreaPayload(testsupport.TestOversizedNameObj)
-
-	err := ci.Validate()
-	// Validate payload function returns an error
-	assert.NotNil(rest.T(), err)
-	assert.Contains(rest.T(), err.Error(), "length of type.name must be less than or equal to 63")
-}
-
-func (rest *TestAreaREST) TestFailValidationAreaNameStartWith() {
-	// given
-	ci := newCreateChildAreaPayload("_TestSuccessCreateChildArea")
-
-	err := ci.Validate()
-	// Validate payload function returns an error
-	assert.NotNil(rest.T(), err)
-	assert.Contains(rest.T(), err.Error(), "type.name must match the regexp")
-}
-
-func (rest *TestAreaREST) TestFailShowAreaNotFound() {
-	// given
-	svc, ctrl := rest.SecuredController()
-	// when/then
-	test.ShowAreaNotFound(rest.T(), svc.Context, svc, ctrl, uuid.NewV4().String(), nil, nil)
-}
-
-func (rest *TestAreaREST) TestShowAreaOK() {
-	// given
-	_, a := createSpaceAndArea(rest.T(), rest.db)
-	svc, ctrl := rest.SecuredController()
-	// when
-	res, _ := test.ShowAreaOK(rest.T(), svc.Context, svc, ctrl, a.ID.String(), nil, nil)
-	//then
-	assertResponseHeaders(rest.T(), res)
-}
+func (rest *TestAreaREST) TestCreateChildArea() {
+	rest.T().Run("Success", func(t *testing.T) {
+		t.Run("OK", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.Identities(2), tf.Areas(1))
+			parentArea := fxt.Areas[0]
+			parentID := parentArea.ID
+			ca := newCreateChildAreaPayload("TestSuccessCreateChildArea")
+			owner := fxt.Identities[0]
+			svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+			// when
+			resp, created := test.CreateChildAreaCreated(t, svc.Context, svc, ctrl, parentID.String(), ca)
+			// then
+			assert.Equal(t, *ca.Data.Attributes.Name, *created.Data.Attributes.Name)
+			assert.Equal(t, parentID.String(), *created.Data.Relationships.Parent.Data.ID)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.res.golden.json"), created)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.res.headers.golden.json"), resp.Header())
+			// try creating child area with different identity: should fail
```

Comment:
Why is this test inside the "ok" section?

Reply:
Yes. It should be in "failure". Please see https://github.com/fabric8-services/fabric8-wit/pull/1941/commits/9444077af47ac799ed6063e48fb0174b1cf9c526